pkg/cache/nodes.go

```diff
@@ -111,8 +162,10 @@ func (nc *NodeController) updateNode(oldObj, newObj interface{}) {
 	}
 }
 
-func (nc *NodeController) deleteNode(obj interface{}) {
-	log.Logger.Debug("node-controller: DeleteNode")
+func (nc *schedulerNodes) deleteNode(obj interface{}) {
+	nc.lock.Lock()
```

Comment:
Can we move the locking down to the point were we need it to make the lock cycle as short as possible?
Just before we call RemoveNode()